keras_retinanet/models/resnet.py

```diff
@@ -80,7 +80,7 @@ def resnet_retinanet(num_classes, backbone=50, inputs=None, weights='imagenet',
 
     # optionally load weights
     if weights_path:
-        model.load_weights(weights_path, by_name=True)
+        model.load_weights(weights_path, by_name=True, skip_mismatch=True)
```

Comment:
Shouldn't we leave this behaviour overridable? We can default it to true, but if you're sure that the weights match the model (for example because you just trained it), not skipping mismatched weights does give some additional error checking.

Reply:
Hmm you raise a valid point. The most common case would be, I expect, that you would intentionally use a mismatching model (for example by always starting with the pretrained COCO model when training a custom dataset). Therefore the default in my opinion should at least be `skip_mismatch=True`. I'm not sure if it then makes sense to allow an option to set `skip_mismatch=False`, but there's no harm in adding it.

Reply:
> The most common case would be, I expect, that you would intentionally use a mismatching model (for example by always starting with the pretrained COCO model when training a custom dataset).

Is this function also used when testing/evaluating? Or is it more common to load the model and weights directly from a file without building the model first?

Reply:
> Is this function also used when testing/evaluating? Or is it more common to load the model and weights directly from a file without building the model first?

It can be used for testing / evaluation, but I would advise against it. Sometimes it can prove useful, for example if there was a small bug in the architecture, with which a network was trained. You can then create a new model with a fixed architecture and load in the weights using this function.

Overall, I would say this function should only be used for training.

Reply:
Right, then I agree defaulting to true makes sense.